state/protocol/errors.go

```diff
@@ -23,7 +23,13 @@ var (
 	ErrNextEpochNotCommitted = fmt.Errorf("queried info from EpochCommit event before it was emitted")
 
 	// ErrEpochTransitionNotFinalized is a sentinel returned when a query is made
-	// for a block at an epoch boundary which has not yet been finalized.
+	// for an epoch boundary which has not been finalized by this node.
+	// We say an epoch boundary between epoch N and M=N+1 is finalized when the first
+	// block within epoch M is finalized.
+	// There are two cases where this can occur:
+	//   1. Epoch M is in the future and no block has been finalized within it yet.
+	//   2. Epoch M is in the past, but its boundary occurred prior to the start of this node's local history.
+	// TODO add badger/state test cases
 	ErrEpochTransitionNotFinalized = fmt.Errorf("cannot query block at un-finalized epoch transition")
```

Comment:
This is a very good point, thank you for bringing it up. I agree the way the error is named and described now is misleading. 

I also agree it would be optimal to use two distinct errors (one for "boundary not finalized yet" and one for "boundary before the beginning of my local state"). I spent some time investigating how we can implement this. I think it is doable but introduces some edge cases which worry me time- and complexity-wise.

For example:
```
LEGEND:
<--> = sealing segment of the root snapshot
*    = reference block of a queried Snapshot
|    = epoch boundary

A: --|---<---->----*
B: --|---<---->---|*
```

Suppose we want to return `PreviousEpoch.FinalHeight` in each case, and **in both cases the reference block is un-finalized**. The right error for Snapshot A is "boundary before the beginning of my local state". The right error for Snapshot B is "boundary not finalized yet". Distinguishing between these cases is tricky and will take some time to get right.

Given that this sentinel is used once in the codebase and for that usage we don't need to distinguish between these cases, I'm inclined to stick with one error type for the `*Height` functions, but improve the error naming and documentation to more clearly describe what it is indicating: that our local state doesn't know about this epoch boundary.